packages/manager/src/queries/cloudpulse/alerts.ts

```diff
@@ -33,9 +33,15 @@ export const useCreateAlertDefinition = (serviceType: AlertServiceType) => {
   return useMutation<Alert, APIError[], CreateAlertDefinitionPayload>({
     mutationFn: (data) => createAlertDefinition(data, serviceType),
     onSuccess: async (newAlert) => {
-      await queryClient.invalidateQueries({
-        queryKey: queryFactory.alerts._ctx.all._def,
-      });
+      const allAlertsKey = queryFactory.alerts._ctx.all().queryKey;
+      const oldAlerts = queryClient.getQueryData<Alert[]>(allAlertsKey);
+
+      if (oldAlerts && oldAlerts.length > 0) {
```

Comment:
```suggestion
      oldAlerts
```

don't check for empty case because it'll be undefined only when it is not fetched, it can have empty data if there are not alerts created by user